### PR TITLE
Adding KeyManager: stage 1

### PR DESCRIPTION
**Pull request by nfurfaro**
# Description
The first step in laying the groundwork for the new keyManager title. This adds some of the wiring, but doesn't actually start using the KM yet.

# Issues

Fixes #
Refs #5557

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [x] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by nfurfaro** [comment]
Actually, it's not that bad.

File: smart-contracts/contracts/mixins/MixinKeys.sol [hunk 1]
```diff
@@ -210,6 +218,31 @@ contract MixinKeys is
...
+  function setKeyManagerOf(
+    uint _tokenId,
+    address _keyManager
+  ) public
+    isKey(_tokenId)
+  {
+    require(
+      keyManagerOf[_tokenId] == msg.sender ||
+      isLockManager(msg.sender) ||
+      keyManagerOf[_tokenId] == address(0), 'UNAUTHORIZED_KEY_MANAGER_UPDATE'
```
Comment (NickCuso) [inline smart-contracts/contracts/mixins/MixinKeys.sol]: Can merge this with the next require with 
``` ... ||
(keyManagerOf[_tokenId] == address(0) && isKeyOwner(...)), 'UNAUTH...
```

File: smart-contracts/test/Lock/permissions/setKeyManager.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+const { reverts } = require('truffle-assertions')
+const BigNumber = require('bignumber.js')
+// const { ZERO_ADDRESS } = require('hardlydifficult-ethereum-contracts')
```
Comment (NickCuso) [inline smart-contracts/test/Lock/permissions/setKeyManager.js]: This is under `constants`

so `const {constants} = require('hd..
constants.ZERO_ADDRESS`

But inline is fine too

File: smart-contracts/contracts/mixins/MixinKeys.sol (hunk 1 above)
Comment (nfurfaro) [reply smart-contracts/contracts/mixins/MixinKeys.sol]: yeah, that works for me. will change it.

File: smart-contracts/test/Lock/permissions/setKeyManager.js (hunk 2 above)
Comment (nfurfaro) [reply smart-contracts/test/Lock/permissions/setKeyManager.js]: May as well use constants, seeing as we have it available already. Will tweak this and push once more.

**Pull request merged**